unit/unit.go

```diff
@@ -58,13 +62,26 @@ func RunUnitTests(fixture *UnitTestFixture) {
 	defer terraform.WorkspaceSelectOrNew(fixture.GoTest, fixture.TfOptions, startingWorkspaceName)
 
 	tfPlanFilePath := filepath.FromSlash(fmt.Sprintf("%s/%s.plan", os.TempDir(), random.UniqueId()))
-	terraform.RunTerraformCommand(
+	defer os.Remove(tfPlanFilePath)
+
+	output, err := terraform.RunTerraformCommandE(
 		fixture.GoTest,
 		fixture.TfOptions,
 		terraform.FormatArgs(fixture.TfOptions, "plan", "-input=false", "-out", tfPlanFilePath)...)
-	defer os.Remove(tfPlanFilePath)
+	if err != nil {
```

Comment:
I think we have some edge cases here:

Ideally we want:
 * `err == nil`, `CommandStdoutAssertions == nil` --> run `validateTerraformPlanFile`
 * `err != nil`, `CommandStdoutAssertions == nil` --> fail
 * `err == nil`, `CommandStdoutAssertions != nil` --> fail
 * `err != nil`, `CommandStdoutAssertions != nil` --> run `validateTerraformCommandStdout`
 
 This implementation does the following, though:
 * `err == nil` --> run `validateTerraformPlanFile`. If `CommandStdoutAssertions != nil`, these assertions are ignored
 * `err != nil` --> run `validateTerraformCommandStdout`. If `CommandStdoutAssertions == nil` then we always pass

Reply:
Can we improve the conditions checking here to cover all cases correctly?

Reply:
Your right there are some cases missing!

However, for this one:
> * `err == nil`, `CommandStdoutAssertions != nil` --> fail

It think it should not fail but run both `validateTerraformPlanFile` & `validateTerraformCommandStdout`. Someone might want to test the command output even if it didn't failed. Checking for warnings or deprecation notice maybe?

What do you think?

Reply:
That's a good point, and I agree!